### PR TITLE
Season stats da

### PR DESCRIPTION
**Pull request by d-atkins**
- Add season stats class
- Add head_to_head (iteration 5)
- Add helper to game which returns the opponent of a team_id

### CODE REVIEW

File: test/game_test.rb [hunk 1]
```diff
@@ -3,24 +3,38 @@
 
 class GameTest < Minitest::Test
   def setup
-    @game = Game.new({season: "20122013", away_goals: 3, home_goals: 4})
+    @game = Game.new({
```
Comment (tylertomlinson) [inline test/game_test.rb:6]: Nice update. Works will all tests now

File: lib/game.rb [hunk 2]
```diff
@@ -19,4 +19,9 @@ def difference_between_score
     difference_between_score = @away_goals - @home_goals
     difference_between_score.abs
   end
+
+  def opponent_id(team_id)
+    return home_team_id if team_id != home_team_id
+    return away_team_id
+  end
```
Comment (Yetidancer) [inline lib/game.rb:26]: hey hey its opposite day

File: test/season_stats_test.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  def setup
+    @game_collection = GameCollection.new("./test/fixtures/games_truncated.csv")
+    @game_teams_collection = GameTeamsCollection.new("./test/fixtures/game_teams_truncated.csv")
+    @season_stats = SeasonStats.new(@game_collection, @game_teams_collection)
+  end
+
+  def test_it_exists
+    assert_instance_of SeasonStats, @season_stats
+  end
+end
```
Comment (Yetidancer) [inline test/season_stats_test.rb]: super looking forward to working in this class

File: lib/game.rb (hunk 2 above)
Comment (d-atkins) [reply lib/game.rb:26]: no

File: lib/season_stats.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  def head_to_head_ids(team_id)
+    results_by_opponents_hash = results_by_opponents(team_id)
+    results_by_opponents_hash.reduce({}) do |acc, results|
+      win_count = results[1].count {|result| result == "WIN"}
+      acc[results[0]] = (win_count.to_f / results_by_opponents_hash[results[0]].length).round(2)
+      acc
+    end
+  end
```
Comment (mronauli) [inline lib/season_stats.rb:25]: awesome!!

File: lib/season_stats.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    @game_collection = game_collection
+    @game_teams_collection = game_teams_collection
+  end
+
+  def results_by_opponents(team_id)
+    @game_teams_collection.game_teams_by_id[team_id].reduce({}) do |acc, game_team|
+      corresponding_game = @game_collection.games.find {|game| game.game_id == game_team.game_id }
+      opponent = corresponding_game.opponent_id(team_id)
+      acc[opponent] << game_team.result if acc[opponent]
+      acc[opponent] = [game_team.result] if acc[opponent].nil?
```
Comment (Yetidancer) [inline lib/season_stats.rb:13]: return hash with opponent keys and result value array

File: test/season_stats_test.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    assert_instance_of SeasonStats, @season_stats
+  end
+
+  def test_it_can_find_results_by_opponent
+    expected = {
+      14=>["WIN", "WIN", "LOSS", "WIN", "WIN", "WIN"],
+      19=>["WIN", "WIN", "WIN", "LOSS", "WIN", "WIN"]
+    }
+    assert_equal expected, @season_stats.results_by_opponents(16)
+  end
```
Comment (mronauli) [inline test/season_stats_test.rb:23]: i like it!

**Pull request merged**